pl_bolts/models/detection/yolo/yolo_config.py

```diff
@@ -264,8 +255,10 @@ def _create_yolo(config, num_inputs):
         overlap_loss_func = yolo_layers.SELoss()
     elif overlap_loss_name == "giou":
         overlap_loss_func = yolo_layers.GIoULoss()
-    else:
+    elif overlap_loss_name == "iou":
         overlap_loss_func = yolo_layers.IoULoss()
+    else:
+        assert False, "Unknown overlap loss: " + overlap_loss_name
```

Comment:
According to the code six lines above, the default should be `mse`

Reply:
This was based on the original YOLO implementation, which I think uses MSE loss. That's why "mse" was the default. Darknet configuration files allow other variants of the iou loss (diou, ciou) that were not available in Torchvision at the time of writing this code, so I chose to use IoULoss in case something else than "mse" or "giou" is specified in the configuration file. I created [another pull request](https://github.com/Lightning-AI/lightning-bolts/pull/817) a while ago that adds support for the ciou and diou losses, that are now available in Torchvision.